extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts

```diff
@@ -292,6 +296,7 @@ export class PublishDatabaseDialog {
 			this.deploymentOptions = await utils.getDefaultPublishDeploymentOptions(this.project) as DeploymentOptions;
 		}
 
+		this.deploymentOptions.optionsMapTable = JSON.parse(JSON.stringify(Object.fromEntries(this.deploymentOptions.optionsMapTable)));
```

Comment:
what is this line doing?

Reply:
Converting the optionsMapTable of Map into JSON object to send it to STS, only if converted into JSON object, i'm able to see the updated values transferred to STS, otherwise nothing coming up there.

Reply:
it doesn't seem right that the type of optionsMapTable is changing